File: nim/atbash-cipher/atbash_cipher.nim
import algorithm, sequtils, strutils, std/wordwrap

const groupSize = 5
const letters = toSeq('a'..'z')
const digits = toSeq('0'..'9')
const plain = letters & digits
const cipher = letters.reversed & digits

proc chunked(phrase: string): string =
  phrase.wrapWords(groupSize, newLine = " ")

proc encode(letter: char): char =
  cipher[plain.find(letter)]

proc encode*(phrase: string): string =
  phrase.toLowerAscii.filter(isAlphaNumeric).map(encode).join.chunked

proc decode(letter: char): char =
  plain[cipher.find(letter)]

proc decode*(phrase: string): string =
  phrase.replace(" ").map(decode).join

File: kotlin/atbash-cipher/src/main/kotlin/AtbashCipher.kt
object Atbash {
    private val letters = 'a'..'z'
    private val encodingMap = letters.zip(letters.reversed()).toMap()

    fun encode(input: String) =
        input.toLowerCase()
            .filter(Char::isLetterOrDigit)
            .chunked(5)
            .map { chunk -> chunk.map { it.encode() }.joinToString("") }
            .joinToString(separator = " ")

    fun decode(input: String) = encode(input).filter(Char::isLetterOrDigit)

    private fun Char.encode() = encodingMap.getOrElse(this) { this }
}